semtypes/src/main/java/io/ballerina/semtype/definition/Field.java

```diff
@@ -34,4 +39,20 @@
     public static Field from(String name, SemType type) {
         return new Field(name, type);
     }
+
+    public static SplitFieldHolder splitFields(List<Field> fields) {
+        Field[] sortedFields = fields.toArray(new Field[0]);
+        Arrays.sort(sortedFields, Comparator.comparing(Field::fieldName));
+        List<String> names = new ArrayList<>();
+        List<SemType> types = new ArrayList<>();
+        for (Field field : sortedFields) {
+            names.add(field.name);
+            types.add(field.type);
+        }
+        return SplitFieldHolder.from(names, types);
+    }
+
+     private static synchronized String fieldName(Field f) {
```

Comment:
`isolated` functions are not just about locks, we need a lock when accessinga module variable. We don't need it here